src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -232,10 +234,24 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 			if (this.cellModel.currentMode !== CellEditModes.WYSIWYG) {
 				needsTransform = false;
 			} else {
+				let result = undefined;
+				const isFile = URI.parse(linkCalloutResult?.insertUnescapedLinkUrl).scheme.includes('file');
+				if (!path.isAbsolute(linkCalloutResult?.insertUnescapedLinkUrl) && isFile) {
+					// Counts the number of ..\ intermediate directories when user enters a relative path
+					let midDirectories = (linkCalloutResult?.insertUnescapedLinkUrl.match(/..\\/g) || []).length;
+					let dirName = path.dirname(this.cellModel?.notebookModel?.notebookUri.fsPath);
+					const relativePath = linkCalloutResult?.insertUnescapedLinkUrl.replace(/\.\\/g, '/');
+					while (midDirectories > 1) {
+						dirName = path.dirname(dirName);
+						midDirectories--;
+					}
+					result = path.join(dirName, relativePath);
+				}
 				// Otherwise, re-focus on the output element, and insert the link directly.
 				this.output?.nativeElement?.focus();
+				result = result ? result : linkCalloutResult?.insertUnescapedLinkUrl;
 				// Callout is responsible for returning escaped strings
-				document.execCommand('insertHTML', false, `<a href="${linkCalloutResult?.insertUnescapedLinkUrl}">${linkCalloutResult?.insertUnescapedLinkLabel}</a>`);
+				document.execCommand('insertHTML', false, `<a href="${result}">${linkCalloutResult?.insertUnescapedLinkLabel}</a>`);
```

Comment:
Shouldn't we be escaping the result and label here? Those might contain HTML characters

Reply:
Can I use `encodeURI` instead of escape in the result? If I use escape then urls are identified as files.